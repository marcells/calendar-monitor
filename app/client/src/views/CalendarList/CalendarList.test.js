import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ShallowRenderer from 'react-test-renderer/shallow';
import ConnectedCalendarList, { CalendarList } from './CalendarList';

const mockStore = configureStore();
const store = mockStore({
  calendars: {
    items: []
  }
});

const mockedDispatch = jest.fn();

describe('CalendarList', () => {
  describe('has items', () => {
    it('should render calendar items correctly', () => {
      const items = [
        {
          id: 'calendar1',
          description: 'some description for calendar 1'
        },
        {
          id: 'calendar2',
          description: 'some description for calendar 2'
        }
      ];

      const renderer = new ShallowRenderer();
      const result = renderer.render(<CalendarList items={items} />);

      expect(result).toMatchSnapshot();
    });
  });

  describe('has no items', () => {
    it('should render calendar items correctly', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(<CalendarList items={[]} />);

      expect(result).toMatchSnapshot();
    });

    it('should render an empty unordered list', () => {
      const wrapper = shallow(<CalendarList store={store} dispatch={mockedDispatch} items={[]} />);

      expect(wrapper.find('ul')).toHaveLength(1);
      expect(wrapper.find('li')).toHaveLength(0);
    });
  });
});
