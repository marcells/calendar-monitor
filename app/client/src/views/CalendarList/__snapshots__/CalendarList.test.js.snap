// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarList has items should render calendar items correctly 1`] = `
<ul>
  <li>
    <Link
      replace={false}
      to="/calendar/calendar1"
    >
      some description for calendar 1
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/calendar/calendar2"
    >
      some description for calendar 2
    </Link>
  </li>
</ul>
`;

exports[`CalendarList has no items should render calendar items correctly 1`] = `
<ul>
  
</ul>
`;
